services/app-api/src/resolvers/rate/generateRateCertificationName.ts

```diff
@@ -1,30 +1,38 @@
 import { formatRateNameDate } from '../../../../app-web/src/common-code/dateHelpers'
-import { packageName } from '../../../../app-web/src/common-code/healthPlanFormDataType'
+import { programNames } from '../../../../app-web/src/common-code/healthPlanFormDataType'
 import type { ProgramArgType } from '../../../../app-web/src/common-code/healthPlanFormDataType'
 import type { RateFormEditableType } from '../../domain-models/contractAndRates'
 
+const naturalSort = (a: string, b: string): number => {
+    return a.localeCompare(b, 'en', { numeric: true })
+}
+
 const generateRateCertificationName = (
     rateFormData: RateFormEditableType,
     stateCode: string,
-    stateNumber: number,
     statePrograms: ProgramArgType[]
 ): string => {
     const {
         rateType,
-        rateProgramIDs,
-        amendmentEffectiveDateEnd,
-        amendmentEffectiveDateStart,
         rateDateCertified,
-        rateDateEnd,
         rateDateStart,
+        rateDateEnd,
+        rateProgramIDs = [],
+        amendmentEffectiveDateStart,
+        amendmentEffectiveDateEnd,
     } = rateFormData
+    const pNames = programNames(statePrograms, rateProgramIDs)
```

Comment:
@pearl-truss Quick question about this. I assume that rateFormData. rateProgramIDs would contain either `not a program` or `program` and never both. Is that correct?

Reply:
@JasonLin0991 I'm sorry I'm just seeing this. I believe it should include both because some states will have a mixture